game-core/src/main/java/games/strategy/triplea/ui/panels/map/MapPanel.java

```diff
@@ -813,8 +810,7 @@ private void drawTiles(
       final List<Tile> undrawn) {
     g.translate(-bounds.getX(), -bounds.getY());
     for (final Tile tile : tileManager.getTiles(bounds)) {
-      tile.acquireLock();
-      try {
+      synchronized (tile.getMutex()) {
```

Comment:
Here and in Line 634 I wasn't sure if this was the best move to expose the mutex like this.
It should be pretty safe to do that, it only ensures that the image isn't changed somehow while we perform drawing operations

Reply:
Actually I found a pretty good solution that avoids having a mutex here entirely.
Will create a follow-up PR once this branch is merged.